src/ScriptEditor/NetscriptDefinitions.d.ts

```diff
@@ -5902,25 +5907,186 @@ interface InfiltrationLocation {
+   *
+   * - The returned object is fixed, i.e. it won't be modified after it is
+   *   returned. Call this function again if you want to see what is happening
+   *   at different times.
+   * - There will be a "stage" field which describes the current minigame, or
+   *   if infiltration is in some other state (intro, collect rewards,
+   *   in-between games, etc.)
+   * - In most cases there will be "floors" and "currentFloor" (1-indexed) to
+   *   indicate your overall progress through the game.
+   * - In most cases there will be "timer" to indicate the time remaining (in
+   *   miliseconds) for this stage.
+   *
+   * Requires API access.
+   * @remarks
+   * RAM cost: 5 GB
+   *
+   * @returns a state object, or null if not infiltrating
+   */
+  getState(): object | null;
+
+  /**
+   * Press a key.
+   *
+   * After a delay, this acts just like pressing a key on the keyboard. This
+   * only functions within infiltration minigames, everywhere else it will be
+   * ignored.
+   *
+   * Keys are generally the string value such as "x", "b", " ", etc.  There
+   * are also special keys such as "ArrowLeft", "ArrowRight", "ArrowUp", and
+   * "ArrowDown" (the arrow keys), which are pretty much the only special keys
+   * you will need. In general, unrecognized keys or keys not relevant to the
+   * current minigame will be ignored.
+   *
+   * The amount of delay depends on the difficulty of the infiltration; harder
+   * infils have more delay. There is also a small random noise that is part
+   * of the delay, which also increases as the difficulty increases.
+   * Difficulties < 1 (the "easy", green part of the bar) won't have a random
+   * component.
+   *
+   * This function should be awaited; like hack(), the script cannot
+   * do other NS calls concurrently until the promise resolves.
+   *
+   * Requires API access.
+   * @remarks
+   * RAM cost: 5 GB
+   */
+  pressKey(key: string): Promise<void>;
+
+  /**
+   * Press spacebar.
+   *
+   * This is like {@link Infiltration.pressKey | pressKey}, but it works
+   * synchronously. This is useful for timing-critical minigames, or just as a
+   * simple way to press space. You can still use `pressKey(" ")` if you want to.
+   *
+   * Requires API access.
+   * @remarks
+   * RAM cost: 5 GB
+   */
+  pressSpace(): void;
+
+  /**
+   * Finish infiltration and claim rewards.
+   *
+   * If passed null, this sells the infiltration rewards for money. Otherwise,
+   * it trades it for rep with the given faction. The faction must be valid
+   * (one the player has joined, and can work for normally).
+   *
+   * This only works from the "victory" screen, and does nothing otherwise.
+   * Like usual, the player will be left on the City page after.
+   *
+   * Requires BN11 or SF11.2
+   * @remarks
+   * RAM cost: 15 GB
+   *
+   * @returns The amount of money or rep gained, or 0 on error
+   */
+  claimRewards(faction: FactionName | null): number;
```

Comment:
```suggestion
  claimRewards(faction?: FactionName): number;
```
I think the explicit null is a bit weird here as a player, makes more sense to me as above: omit the faction arg to get the money reward. Doesn't need an actual implementation change as the actual function code treats undefined and null the same.

Would need a doc change as well in the description for the faction parameter.